File: Dental.Client/src/app/models/patient.ts
export enum EducationLevel {
  ElementarySchool = 'ElementarySchool',
  HighSchool = 'HighSchool',
  Graduate = 'Graduate',
  Masters = 'Masters',
  Phd = 'Phd',
}

export interface Patient {
  id: string;
  name: string;
  surname: string;
  socialSecurityNumber: string;
  address?: string | null;
  phone?: string | null;
  email?: string | null;
  dateOfBirth: string;
  gender: string;
  education: EducationLevel;
  bloodType?: string | null;
  emergencyContactName?: string | null;
  emergencyContactPhone?: string | null;
  clinicalSummary?: string | null;
  isActive?: boolean;
}

export interface PatientAllergy {
  id: string;
  patientId: string;
  substance: string;
  severity?: string | null;
  reaction?: string | null;
  isActive: boolean;
}

export interface PatientMedicalHistory {
  id: string;
  patientId: string;
  title: string;
  description?: string | null;
  recordedOn?: string | null;
  recordedBy?: string | null;
}

export interface PatientClinicalNote {
  id: string;
  patientId: string;
  title: string;
  content: string;
  isConfidential: boolean;
}

export interface PatientKvkkConsent {
  id: string;
  patientId: string;
  consentType: string;
  isGranted: boolean;
  consentedAt: string;
  consentVersion?: string | null;
}

export interface PatientDocument {
  id: string;
  patientId: string;
  fileName: string;
  category: string;
  fileSizeBytes: number;
  description?: string | null;
}

export interface PatientBalance {
  patientId: string;
  totalCharges: number;
  totalPayments: number;
  balance: number;
}

File: Dental.Web/Models/Patient.cs
using System.ComponentModel.DataAnnotations;
using Dental.Web.Models.Common;

namespace Dental.Web.Models;

public class Patient : AuditableEntity
{
    public Guid Id { get; set; }

    [Required]
    [MaxLength(128)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(128)]
    public string Surname { get; set; } = string.Empty;

    [Required]
    [MaxLength(32)]
    public string SocialSecurityNumber { get; set; } = string.Empty;

    [MaxLength(512)]
    public string? Address { get; set; }

    [MaxLength(64)]
    public string? Phone { get; set; }

    [MaxLength(256)]
    public string? Email { get; set; }

    public DateOnly DateOfBirth { get; set; }

    [Required]
    [MaxLength(32)]
    public string Gender { get; set; } = string.Empty;

    public EducationLevel Education { get; set; }

    [MaxLength(16)]
    public string? BloodType { get; set; }

    [MaxLength(128)]
    public string? EmergencyContactName { get; set; }

    [MaxLength(64)]
    public string? EmergencyContactPhone { get; set; }

    [MaxLength(2000)]
    public string? ClinicalSummary { get; set; }

    public string? UserId { get; set; }

    public ApplicationUser? User { get; set; }

    [MaxLength(64)]
    public string? RegistrationInviteToken { get; set; }

    public DateTimeOffset? RegistrationInviteExpiresAt { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<PatientMedicalHistory> MedicalHistories { get; set; } = [];
    public ICollection<PatientClinicalNote> ClinicalNotes { get; set; } = [];
    public ICollection<PatientAllergy> Allergies { get; set; } = [];
    public ICollection<PatientChronicCondition> ChronicConditions { get; set; } = [];
    public ICollection<PatientDocument> Documents { get; set; } = [];
    public ICollection<PatientKvkkConsent> KvkkConsents { get; set; } = [];
}
